packages/library-base/src/commonMain/kotlin/io/realm/query/RealmQuery.kt

```diff
@@ -246,15 +250,27 @@ interface RealmSingleQuery<T : RealmObject> {
      * Finds the first object that fulfills the query conditions and returns it asynchronously as a
      * [Flow].
      *
-     * If there is any changes to the object represented by the query, the flow will emit the
-     * updated object. The flow will continue running indefinitely until canceled.
+     * If there is any changes to the first object represented by the query, the flow will emit an
+     * [ObjectChange] event depending on the query state. The flow will continue running indefinitely until
+     * canceled.
+     *
+     * If subscribed on an empty query the flow will not yield any event until an element is added, then
```

Comment:
Hmm, I'm starting to think that we might have to rethink semantics around query types and data model types....I should probably have thought of this sooner 🤕 

We have two types of flows on queries:

1) On RealmResults
2) On RealmSingleQuery

Both of these differ from the states we assigned to objects, lists and eventually map and set.

* For RealmResults, the difference is that the deleted state doesn't really make sense
* For SingleQuery, the difference is that the initial state might be empty.

We can probably reuse the types we already have (which is what you are doing), but I'm beginning to wonder if that is too annoying to consume or we should introduce two new Change types: `QueryChange` and `ObjectQueryChange` (or something like that)....not sure if that is something you have thought about?